packages/2018-neighborhood-development/src/components/ExploreUrbanCampsiteSweeps/index.js

```diff
@@ -77,24 +79,25 @@ export class ExploreUrbanCampsiteSweeps extends React.Component {
 ExploreUrbanCampsiteSweeps.displayName = 'ExploreUrbanCampsiteSweeps';
 ExploreUrbanCampsiteSweeps.propTypes = {
   init: PropTypes.func,
+  play: PropTypes.func,
   isLoading: PropTypes.bool,
   error: PropTypes.string,
   data: PropTypes.arrayOf(PropTypes.object),
+  timer: PropTypes.int,
 };
 
 export default connect(
 
   state => ({
     isLoading: isCampsiteSweepsPending(state),
     error: catchCampsiteSweepsErrors(state),
-    data: getCampsiteSweepsData(state),
+    data: getCampsiteSweepsDataByTime(state),
+    timer: getCampsiteSweepsDateRange(state),
   }),
   dispatch => ({
     init() {
       dispatch(fetchCampsiteSweeps());
-    },
-    incrementTimer() {
-      dispatch(incrementTimer(5));
+      setInterval(() => dispatch(incrementTimer(1)), 800);
```

Comment:
Calling `setInterval` like this may cause a memory leak. I suspect you will have to clear the interval when the component is unmounted. Something like...

```js
class Card extends React.Component {
  componentDidMount() {
    const timer = setInterval(() => { this.props.tick(); }, 800);
    this.setState({ timer });
  },

  componentWillUnmount() {
    clearInterval(this.state.timer);
  },
}

export default connect({
  state => ...,
  dispatch => ({
    tick() {
      dispatch(incrementTimer(1));
    }
  })
});
```

This means the component is stateful but at least only the very local timer state is stored on the component instead of in the Redux store. Other possibilities: https://medium.com/@machadogj/timers-in-react-with-redux-apps-9a5a722162e8

Reply:
Did essentially this, you're only storing an ID of the interval you set in state, so the state is very minimal. Thanks for the help.